pandas/core/ops.py

```diff
@@ -654,20 +654,48 @@ def na_op(x, y):
         return result
 
     def wrapper(self, other):
+        is_self_int_dtype = com.is_integer_dtype(self.dtype)
+
         if isinstance(other, pd.Series):
             name = _maybe_match_name(self, other)
 
-            other = other.reindex_like(self).fillna(False).astype(bool)
-            return self._constructor(na_op(self.values, other.values),
+            other = other.reindex_like(self)
+            is_other_int_dtype = com.is_integer_dtype(other.dtype)
+
+            if is_other_int_dtype:
+                other = other.fillna(0)
+            else:
+                other = other.fillna(False).astype(bool)
+
+            res = self._constructor(na_op(self.values, other.values),
                                      index=self.index,
-                                     name=name).fillna(False).astype(bool)
+                                     name=name)
+            if is_self_int_dtype and is_other_int_dtype:
+                res = res.fillna(0)
+            else:
+                res = res.fillna(False).astype(bool)
+
+            return res
+
         elif isinstance(other, pd.DataFrame):
             return NotImplemented
+
         else:
-            # scalars
+            # scalars, list, tuple, np.array
             res = self._constructor(na_op(self.values, other),
-                                    index=self.index).fillna(False)
-            return res.astype(bool).__finalize__(self)
+                                    index=self.index)
+
+            if isinstance(other, (list, tuple, np.ndarray)) and \
+                all(com.is_integer(x) for x in other) and is_self_int_dtype:
```

Comment:
instead of doing this list-comp just do (you can incorporate in what I show above as well)

```
if isinstance(other, (list,tuple,np.ndarray)) and is_self_int_dtype:
    if com.is_integer_dtype(np.array(other,copy=False)):
        res = res.fillna(0)
```